File: src/main/java/net/minestom/server/attribute/AttributeOperation.java
package net.minestom.server.attribute;

import org.jetbrains.annotations.Nullable;

public enum AttributeOperation {
    ADDITION(0),
    MULTIPLY_BASE(1),
    MULTIPLY_TOTAL(2);

    private static final AttributeOperation[] VALUES = new AttributeOperation[]{ADDITION, MULTIPLY_BASE, MULTIPLY_TOTAL};
    private final int id;

    AttributeOperation(int id) {
        this.id = id;
    }

    public int getId() {
        return this.id;
    }

    public static @Nullable AttributeOperation fromId(int id) {
        if (id >= 0 && id < VALUES.length) {
            return VALUES[id];
        }
        return null;
    }
}

File: src/main/java/net/minestom/server/attribute/AttributeOperation.kt
package net.minestom.server.attribute

import java.lang.IllegalArgumentException
import java.util.concurrent.ConcurrentHashMap
import net.minestom.server.attribute.AttributeInstance
import net.minestom.server.attribute.AttributeModifier
import net.minestom.server.attribute.AttributeOperation
import java.util.function.IntFunction

enum class AttributeOperation(val id: Int) {
    ADDITION(0), MULTIPLY_BASE(1), MULTIPLY_TOTAL(2);

    companion object {
        private val VALUES = arrayOf(ADDITION, MULTIPLY_BASE, MULTIPLY_TOTAL)
        @JvmStatic
        fun fromId(id: Int): AttributeOperation? {
            return if (id >= 0 && id < VALUES.size) {
                VALUES[id]
            } else null
        }
    }
}